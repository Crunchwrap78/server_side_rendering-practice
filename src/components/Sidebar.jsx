import React, { Component } from 'react';

export default class Sidebar extends Component {
    constructor() {
      super()
      this.cities = this.cities.bind(this)
      this.homeTypes = this.homeTypes.bind(this)
      this.bedrooms = this.bedrooms.bind(this)
    }
    componentWillMount() {
      this.props.populateOption();
    }

    cities() {
      if(this.props.globalState.populateDropDown.cities != undefined) {
        var { cities } = this.props.globalState.populateDropDown
        return cities.map(item => <option key={item} value={item}>{item}</option>)
      }
    }

    homeTypes() {
      if(this.props.globalState.populateDropDown.homeTypes != undefined) {
        var { homeTypes } = this.props.globalState.populateDropDown
        return homeTypes.map(item => <option key={item} value={item}>{item}</option>)
      }
    }

    bedrooms() {
      if(this.props.globalState.populateDropDown.numOfRooms != undefined) {
        var { numOfRooms } = this.props.globalState.populateDropDown
        return numOfRooms.map(item => <option key={item} value={item}>{item} bedrooms </option>)
      }
    }
    render() {
      return (
        <section id="sidebar">
          <div className="inside">
            <h4>Filter</h4>
            <select name="city" className="filters neighborhood" onChange={this.props.handleChange}>
              <option>All</option>
              {this.cities()}
            </select>
            <select name="homeType" className="filters housetype" onChange={this.props.handleChange}>
              <option>All</option>
              {this.homeTypes()}
            </select>
            <select name="rooms" className="filters bedrooms" onChange={this.props.handleChange}>
              {this.bedrooms()}
            </select>
            <div className="filters price">
              <span className="title">Price</span>
              <input type="text" name="min_price" className="min-price" onChange={this.props.handleChange} value={this.props.globalState.min_price}/>
              <input type="text" name="max_price" className="max-price" onChange={this.props.handleChange} value={this.props.globalState.max_price}/>
            </div>
            <div className="filters floor-space">
              <span className="title">Floor Space </span>
              <input type="text" name="min_floor_space" className="min-floor-space" onChange={this.props.handleChange} value={this.props.globalState.min_floor_space}/>
              <input type="text" name="max_floor_space" className="max-floor-space" onChange={this.props.handleChange} value={this.props.globalState.max_floor_space}/>
            </div>
            <div className="filters extras">
              <span className="title">Extras</span>
              <label htmlFor="extras">
                <span>Elevators</span>
                <input name="elevator" value="elevator" type="checkbox" onChange={this.props.handleChange}/>
              </label>
              <label htmlFor="extras">
                <span>Swimming Pool</span>
                <input name="swimming_pool" value="swimming_pool" type="checkbox" onChange={this.props.handleChange}/>
              </label>
              <label htmlFor="extras">
                <span>Basement</span>
                <input name="basement" value="basement" type="checkbox" onChange={this.props.handleChange}/>
              </label>
              <label htmlFor="extras">
                <span>Gym</span>
                <input name="gym" value="gym" type="checkbox" onChange={this.props.handleChange}/>
              </label>
            </div>
          </div>
        </section>
      );
   }
}
