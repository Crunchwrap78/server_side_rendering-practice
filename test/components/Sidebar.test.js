import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from '../../src/components/Sidebar';
import listingsData from '../../src/data/listings.js'

let handleChange, filterData, populateOption, wrapper;

beforeEach(() => {
  const globalState = {
    city: 'All',
    rooms: 0,
    listingsData,
    homeType: 'All',
    min_price: 0,
    max_price: 10000000,
    min_floor_space: 0,
    max_floor_space: 4000,
    elevator: false,
    basement: false,
    swimming_pool: false,
    gym: false,
    filteredData: listingsData,
    populateDropDown: '',
    sortby: 'price-dsc',
    search: '',
    view: 'long'
  };
  handleChange = jest.fn();
  filterData = jest.fn();
  populateOption = jest.fn();
  wrapper = shallow(
    <Sidebar
      handleChange={handleChange}
      filterData={filterData}
      populateOption={populateOption}
      globalState={globalState}
    />
  );
});

test('should render Sidebar correctly', () => {
  expect(wrapper).toMatchSnapshot();
  expect(populateOption).toHaveBeenCalled();
});
test('should handle sort cities', () => {
  const value = 'Condo';
  wrapper.find('select').at(0).simulate('change', {
    target: { value }
  });
  expect(handleChange).toHaveBeenCalled();
});
test('should handle sort homeType', () => {
  const value = 'Miami';
  wrapper.find('select').at(1).simulate('change', {
    target: { value }
  });
  expect(handleChange).toHaveBeenCalled();
});
test('should handle sort price', () => {
  const value = 'price-asc';
  wrapper.find('select').at(2).simulate('change', {
    target: { value }
  });
  expect(handleChange).toHaveBeenCalled();
});
