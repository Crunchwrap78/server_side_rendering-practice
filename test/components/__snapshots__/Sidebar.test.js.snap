// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sidebar correctly 1`] = `
<section
  id="sidebar"
>
  <div
    className="inside"
  >
    <h4>
      Filter
    </h4>
    <select
      className="filters neighborhood"
      name="city"
      onChange={[Function]}
    >
      <option>
        All
      </option>
    </select>
    <select
      className="filters housetype"
      name="homeType"
      onChange={[Function]}
    >
      <option>
        All
      </option>
    </select>
    <select
      className="filters bedrooms"
      name="rooms"
      onChange={[Function]}
    />
    <div
      className="filters price"
    >
      <span
        className="title"
      >
        Price
      </span>
      <input
        className="min-price"
        name="min_price"
        onChange={[Function]}
        type="text"
        value={0}
      />
      <input
        className="max-price"
        name="max_price"
        onChange={[Function]}
        type="text"
        value={10000000}
      />
    </div>
    <div
      className="filters floor-space"
    >
      <span
        className="title"
      >
        Floor Space 
      </span>
      <input
        className="min-floor-space"
        name="min_floor_space"
        onChange={[Function]}
        type="text"
        value={0}
      />
      <input
        className="max-floor-space"
        name="max_floor_space"
        onChange={[Function]}
        type="text"
        value={4000}
      />
    </div>
    <div
      className="filters extras"
    >
      <span
        className="title"
      >
        Extras
      </span>
      <label
        htmlFor="extras"
      >
        <span>
          Elevators
        </span>
        <input
          name="elevator"
          onChange={[Function]}
          type="checkbox"
          value="elevator"
        />
      </label>
      <label
        htmlFor="extras"
      >
        <span>
          Swimming Pool
        </span>
        <input
          name="swimming_pool"
          onChange={[Function]}
          type="checkbox"
          value="swimming_pool"
        />
      </label>
      <label
        htmlFor="extras"
      >
        <span>
          Basement
        </span>
        <input
          name="basement"
          onChange={[Function]}
          type="checkbox"
          value="basement"
        />
      </label>
      <label
        htmlFor="extras"
      >
        <span>
          Gym
        </span>
        <input
          name="gym"
          onChange={[Function]}
          type="checkbox"
          value="gym"
        />
      </label>
    </div>
  </div>
</section>
`;
